File: app/src/main/java/com/example/hustlefix/WelcomeActivity.java
package com.example.hustlefix;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
public class WelcomeActivity extends AppCompatActivity {
    private Button btnServiceProvider, btnClient;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        LanguageManager.applyLanguage(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);
        btnServiceProvider = findViewById(R.id.btnServiceProvider);
        btnClient = findViewById(R.id.btnClient);

        // Global Banner is just decorative/text, no need to link unless dynamic

        btnServiceProvider.setOnClickListener(v -> goToLogin("service_provider"));
        btnClient.setOnClickListener(v -> goToLogin("CLIENT"));
    }
    private void goToLogin(String role) {
        SessionHelper.saveRole(this, role);
        Intent intent = new Intent(WelcomeActivity.this, LoginActivity.class);
        intent.putExtra("ROLE", role);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/hustlefix/WelcomeActivity.kt
package com.example.hustlefix

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import com.example.hustlefix.ui.screens.WelcomeScreen
import com.example.hustlefix.ui.theme.HustleFixTheme

class WelcomeActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        LanguageManager.applyLanguage(this)
        super.onCreate(savedInstanceState)
        
        setContent {
            HustleFixTheme {
                WelcomeScreen(
                    onServiceProviderClick = { goToLogin("service_provider") },
                    onClientClick = { goToLogin("CLIENT") }
                )
            }
        }
    }

    private fun goToLogin(role: String) {
        SessionHelper.saveRole(this, role)
        val intent = Intent(this, LoginActivity::class.java)
        intent.putExtra("ROLE", role)
        startActivity(intent)
    }
}
